featuretools/entityset/serialization.py

```diff
@@ -1,175 +1,228 @@
-# -*- coding: utf-8 -*-
-
 import json
 import os
 import shutil
-import sys
-import uuid
-from tempfile import mkdtemp
 
-import numpy as np
 import pandas as pd
-from pandas.io.pickle import read_pickle as pd_read_pickle
-from pandas.io.pickle import to_pickle as pd_to_pickle
+
+from .. import variable_types
+
+FORMATS = ['csv', 'pickle', 'parquet']
+SCHEMA = {
+    'entities': ['id', 'index', 'time_index', 'variables', 'properties', 'loading_info'],
+    'relationships': {
+        'parent': ['parent_entity', 'parent_variable'],
+        'child': ['child_entity', 'child_variable']
+    }
+}
+VARIABLE_TYPES = {
+    getattr(variable_types, type).type_string: getattr(variable_types, type) for type in dir(variable_types)
+    if hasattr(getattr(variable_types, type), 'type_string')
+}
 
 
-def read_parquet(path, load_data=True):
-    '''Load an EntitySet from a path on disk, assuming
-    the EntitySet was saved in the parquet format.
+def to_entity_description(entity):
```

Comment:
I dont see the advantage of using the `SCHEMA` here. why not just make this function

```
index = entity.df.columns.isin([variable.id for variable in entity.variables])
dtypes = entity.df[entity.df.columns[index]].dtypes.astype(str).to_dict()
description = {
    "id": entity.id,
    "index": entity.index,
    "time_index": entity.time_index,
    "properties": {
        'secondary_time_index': entity.secondary_time_index,
        'last_time_index': entity.last_time_index is not None,
    },
    "variables": [variable.create_data_description() for variable in entity.variables],
    "loading_info": {
        'params': {},
        'properties': {'dtypes': dtypes}
    }
}

return description
```

this is more explicit

Reply:
Change applied in [this commit.](https://github.com/Featuretools/featuretools/pull/361/commits/a642a4db72be52203ab6c05fc9d580c462885389)